contracts/handler/extensible/MarshalLib.sol

```diff
@@ -56,7 +56,7 @@ library MarshalLib {
         assembly {
             // set isStatic to true if the left-most byte of the data is 0x00
             isStatic := iszero(shr(248, data))
-            handler := shr(96, shl(96, data))
+            handler := and(data, 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)
```

Comment:
Hmmm, can the conversion be omitted here at all? Usually, if the evm accesses an address, it only takes the last 20 bytes, ignoring potential dirty bits. Maybe check with the certora?

I did it locally, and no test failed.

Reply:
Reply from Certora:
```
I believe the bits should be cleaned for the addresses as only a reuse in Solidity itself would clean them. If another assembly blocks uses it, the dirty bits will remain. Given that this is a lib, how it's used isn't guaranteed, so it's better to manually clean. See also the warning from the Solidity language documentation. However for bytes4 sigSelector, we're back to solidity after the assembly block (and the value isn't reused later), so it can indeed be removed.
```

![image](https://github.com/user-attachments/assets/e03a5e52-8f7f-4da6-bc4b-a34d3e43beaa)

Reply:
Kept the handler cleaning, and removed the `sigSelector` one. Nice catch!